### DISCUSSION TITLE
VM Sizes

### DISCUSSION THREAD

**Discussion by thepaulmacca** [actions/virtual-environments#3326] [Image configuration]
I'm looking to use these scripts to create Ubuntu2004 and Windows2019 images for some VMSS agents, and was wondering for those who've done the same, what VM sizes you went for, or recommend?

I'm thinking of using **Standard_D4s_v3** for Ubuntu, so I can take advantage of ephemeral OS disks due to the disk size, but I won't get away with those for Windows. What sizes have worked for you for Windows?

Any tips appreciated, thanks

**Comment by JRobertsDS** [comment]
88 GB has worked for us on ubuntu2004.

**Comment by sharadbharadia** [comment]
Hello @thepaulmacca even I am trying to use these scripts to create Ubuntu2004 image, however, the same thing fails while executing 
```
{
        "type": "shell",
        "scripts": [
            "{{template_dir}}/scripts/installers/complete-snap-setup.sh"
        ],
        "environment_vars": [
            "HELPER_SCRIPTS={{user `helper_script_folder`}}"
        ],
        "execute_command": "sudo sh -c '{{ .Vars }} {{ .Path }}'"
},
```

Error message says:

```
1623504910,,ui,say,==> azure-arm: Provisioning with shell script: D:\a\1\s\VMs\linux/scripts/installers/configure-environment.sh
1623504910,,ui,message,    azure-arm: ImageVersion=dev
1623504910,,ui,message,    azure-arm: ImageOS=ubuntu20
1623504910,,ui,message,    azure-arm: ACCEPT_EULA=Y
1623504910,,ui,message,    azure-arm: XDG_CONFIG_HOME=$HOME/.config
1623504910,,ui,message,    azure-arm: AGENT_TOOLSDIRECTORY=/opt/hostedtoolcache
1623504910,,ui,message,    azure-arm: vm.max_map_count=26214
1623504910,,ui,say,==> azure-arm: Provisioning with shell script: D:\a\1\s\VMs\linux/scripts/installers/complete-snap-setup.sh
1623504911,,ui,error,==> azure-arm: /imagegeneration/helpers/etc-environment.sh: line 6: $'\r': command not found
```

Spent quite time to diagnose the issue but couldn't. Any ideas why it is failing ?

ADO pipeline is hosted on `Microsoft Hosted Agent : vmImage: 'windows-2019'` which calls a PowerShell script `Pack-Image.ps1` to create the image.

Appreciate any help here.

**Comment by igorpupkinable** [comment]
I have successfully created Ubuntu 22 images with Standard B2als v2 VM.
I did try with Standard_B1ms as well.

Here is the fork of this repo  with Azure Pipeline configured to build Ubuntu agents
https://github.com/igorpupkinable/azure-pipeline-agent/
